bindings/c#/run.me

```diff
@@ -25,27 +25,55 @@ using System.IO;
 namespace supranational { public static class blst {
 
 #if NET5_0_OR_GREATER
-private static string dll;
+
+private static readonly string dll;
 
 static blst()
 {
-    if (String.IsNullOrEmpty(dll)) {
-        var name = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "blst.dll"
-                 : RuntimeInformation.IsOSPlatform(OSPlatform.OSX)     ? "libblst.dll.dylib"
-                 : "libblst.dll.so";
-
-        var dir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
-        var arch = RuntimeInformation.ProcessArchitecture switch {
-            Architecture.X64   => "x64",
+    if (String.IsNullOrEmpty(dll))
+    {
+        var dir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) ?? ".";
+        var arch = RuntimeInformation.ProcessArchitecture switch
+        {
+            Architecture.X64 => "x64",
             Architecture.Arm64 => "arm64",
             _ => "unsupported"
         };
-        dll = Path.Combine(dir, arch, name);
+        var os = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "win"
```

Comment:
Since you went for `#if NET8_0_OR_GREATER`, this declaration and assignment should be moved to the `#else`, where it's used.